src/main/java/org/isf/users/rest/UserController.java

```diff
@@ -43,215 +45,220 @@
-     * @throws OHServiceException When failed to create user
-     */
-    @ResponseStatus(HttpStatus.CREATED)
-    @PostMapping(value = "/users", produces = MediaType.APPLICATION_JSON_VALUE)
-    public UserDTO newUser(@Valid @RequestBody UserDTO userDTO) throws OHServiceException {
-        LOGGER.info("Attempting to create user {}.", userDTO.getUserName());
-        User user = userMapper.map2Model(userDTO);
-        try {
-            LOGGER.info("User successfully created.");
-            return userMapper.map2DTO(userManager.newUser(user));
-        } catch (OHServiceException serviceException) {
-            LOGGER.info("User is not created.");
-            throw new OHAPIException(new OHExceptionMessage("User not created."));
-        }
-    }
+	/**
+	 * Creates a new {@link User}.
+	 * @param userDTO - the {@link User} to insert
+	 * @return the new created user.
+	 * @throws OHServiceException When failed to create user
+	 */
+	@ResponseStatus(HttpStatus.CREATED)
+	@PostMapping(value = "/users", produces = MediaType.APPLICATION_JSON_VALUE)
+	public UserDTO newUser(@Valid @RequestBody UserDTO userDTO) throws OHServiceException {
+		LOGGER.info("Attempting to create user {}.", userDTO.getUserName());
+		User user = userMapper.map2Model(userDTO);
+		try {
+			LOGGER.info("User successfully created.");
+			return userMapper.map2DTO(userManager.newUser(user));
+		} catch (OHServiceException serviceException) {
+			LOGGER.info("User is not created.");
+			throw new OHAPIException(new OHExceptionMessage("User not created."));
+		}
+	}
 
-    /**
-     * Deletes an existing {@link User}.
-     *
-     * @param username - the name of the {@link User} to delete
-     */
-    @ResponseStatus(HttpStatus.NO_CONTENT)
-    @DeleteMapping(value = "/users/{username}", produces = MediaType.APPLICATION_JSON_VALUE)
-    public void deleteUser(@PathVariable String username) throws OHServiceException {
-        User foundUser = userManager.getUserByName(username);
-        if (foundUser == null) {
-            throw new OHAPIException(new OHExceptionMessage("User not found."));
-        }
-        try {
-            userManager.deleteUser(foundUser);
-        } catch (OHServiceException serviceException) {
-            throw new OHAPIException(new OHExceptionMessage("User not deleted."));
-        }
-    }
+	/**
+	 * Deletes an existing {@link User}.
+	 * @param username - the name of the {@link User} to delete
+	 */
+	@ResponseStatus(HttpStatus.NO_CONTENT)
+	@DeleteMapping(value = "/users/{username}", produces = MediaType.APPLICATION_JSON_VALUE)
+	public void deleteUser(@PathVariable String username) throws OHServiceException {
+		User foundUser = userManager.getUserByName(username);
+		if (foundUser == null) {
+			throw new OHAPIException(new OHExceptionMessage("User not found."), HttpStatus.NOT_FOUND);
+		}
+		try {
+			try {
+				userManager.deleteUser(foundUser);
+			} catch (OHServiceException ex) {
+				foundUser.setDeleted(true);
```

Comment:
This looks strange.  Why would I expect a failed `deleteUser()` method to suddenly succeed by setting the `isDeleted` field? 

I would expect lines 181-183 to be removed along with line 186.

So it would look something like the old code deleted above with a `foundUser.setDeleted(true);` call after the old line 176

Reply:
Hello @dbmalkovsky the aim, at this stage, is to try an hard-deletion and, if not possible, to proceed with a soft-deletion, in order to avoid two different API calls... this is notified to the user in the UI with a message before the call (e.g. "If the user cannot be deleted will be marked as "deleted". Ok to proceed?".
It is not perfect, I agree that we should have two different calls (one for deletion and another one for update, simply) but we can manage it in a later stage if it makes sense... wdyt?

Reply:
@mwithi OK.  I would vote to take the time to do it correctly now but I guess I can see the need to do it this way for now.   Just seems less professional to me.

Reply:
@SteveGT96 don't you think we should use the same pattern as in GUI?

delete -> only delete
update -> only update (even if setting to deleted=yes)

Reply:
> @SteveGT96 don't you think we should use the same pattern as in GUI?
> 
> delete -> only delete update -> only update (soft-deletion is an update)

Actually we are using almost the same pattern on the GUI. When we cannot delete the item due to DataIntegrityException, we make a second request to update it and mark it as deleted.

Please, could you describe very well what you want to be done ? It would be nice to describe also acceptance cases.

Reply:
Well, at the moment you don't actually do a second request but you just manage the update within the same call.
The standard procedure should be that, having the error message in return, the UI makes another call to the update method. If you want to make things consistent it shoud be like that, wdyt?

Reply:
I'd not be evident to rely on the error message to trigger the second call(since the request could faild for another reason and the error messages are not really standardized). That said, I suggest to split that in two actions on the frontend:

- "Delete":  for hard deletion.
- "Mark as deleted": that just update the item and mark it as deleted.

Does it looks OK ?

Reply:
I think it's ok, I didn't mean to trigger the second call automatically, but managed by the user,  for which the update method should be enough